api-tests/server/checks_test.go

```diff
@@ -84,15 +85,41 @@ func TestListSecurityChecks(t *testing.T) {
 	toggleSTT(t, true)
 	t.Cleanup(func() { restoreSettingsDefaults(t) })
 
-	resp, err := managementClient.Default.SecurityChecks.ListSecurityChecks(nil)
-	require.NoError(t, err)
-	assert.NotNil(t, resp)
-	assert.NotEmpty(t, resp.Payload.Checks)
-	for _, c := range resp.Payload.Checks {
-		assert.NotEmpty(t, c.Name, "%+v", c)
-		assert.NotEmpty(t, c.Summary, "%+v", c)
-		assert.NotEmpty(t, c.Description, "%+v", c)
-	}
+	t.Run("with categories filter", func(t *testing.T) {
+		filter := `{"filter_params": {"category": {"string_values": {"values": ["performance"]}}}}`
+		filterParams := &security_checks.ListSecurityChecksParamsBodyFilterParams{}
+		err := json.Unmarshal([]byte(filter), filterParams)
+		assert.NoError(t, err)
+
+		resp, err := managementClient.Default.SecurityChecks.ListSecurityChecks(&security_checks.ListSecurityChecksParams{
+			Body: security_checks.ListSecurityChecksBody{
+				FilterParams: filterParams,
+			},
+		})
+		require.NoError(t, err)
+		assert.NotNil(t, resp)
+		assert.NotNil(t, resp.Payload)
+		if len(resp.Payload.Checks) != 0 {
+			for _, c := range resp.Payload.Checks {
+				assert.NotEmpty(t, c.Name, "%+v", c)
+				assert.NotEmpty(t, c.Summary, "%+v", c)
+				assert.NotEmpty(t, c.Description, "%+v", c)
+				assert.Equal(t, c.Category, "performance")
```

Comment:
At the moment, `string_values` can't take arrays directly because we use `oneOf` in the proto definition and it doesn't allow repeated values (https://developers.google.com/protocol-buffers/docs/proto3#using_oneof),